# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../udev/readdir64.patch
# Copyright (C) 2011 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Compensate for dietlibc missing fields in the legacy dirent, ...

  - Rene Rebe <rene@exactcode.de>

--- udev-165/extras/path_id/path_id.c.vanilla	2011-01-07 09:56:56.659133651 +0000
+++ udev-165/extras/path_id/path_id.c	2011-01-07 09:57:26.747805977 +0000
@@ -189,7 +189,7 @@
 	char *base;
 	char *pos;
 	DIR *dir;
-	struct dirent *dent;
+	struct dirent64 *dent;
 	int basenum;
 
 	hostdev = udev_device_get_parent_with_subsystem_devtype(parent, "scsi", "scsi_host");
@@ -216,7 +216,7 @@
 		parent = NULL;
 		goto out;
 	}
-	for (dent = readdir(dir); dent != NULL; dent = readdir(dir)) {
+	for (dent = readdir64(dir); dent != NULL; dent = readdir64(dir)) {
 		char *rest;
 		int i;
 
--- udev-167/libudev/libudev-device.c.vanilla	2011-05-24 15:28:59.870234223 +0200
+++ udev-167/libudev/libudev-device.c	2011-05-24 15:29:15.510229890 +0200
@@ -1280,7 +1280,7 @@
 
 static int udev_device_sysattr_list_read(struct udev_device *udev_device)
 {
-	struct dirent *dent;
+	struct dirent64 *dent;
 	DIR *dir;
 	int num = 0;
 
